### ISSUE TITLE
[BC Break] "framework.templating.assets_base_urls" in config

### ISSUE THREAD

**Issue by Alsatian67** [symfony/symfony#14332]
To separate the assets of dev and prod environments, I was using different "export" paths.

These paths were configurated twice : For Assetic / For framework.templating in config_prod.yml and config_dev.yml.

My config_dev.yml : 

framework.templating.assets_base_urls: /source

-> All my dev assets were pointing to web/source ...

Since I did the upgrade from Sf 2.6 To 2.7 Beta1, this parameter brings this exception :

InvalidArgumentException in UrlPackage.php line 129: "/source" is not a valid URL.

Is it a BC a break or a bad configuration from me ?

May be it is linked to one of these PR :
[Asset] added the component:
https://github.com/symfony/symfony/pull/13234

Twig decoupling from Templating:
https://github.com/symfony/symfony/pull/13354

**Comment by cordoval** [comment]
can you please try to reproduce this with SE and link here the reproduction? would help a lot to troubleshoot

**Comment by xabbuh** [comment]
`/source` is no URL. With the new Asset component in Symfony 2.7, you can use the `framework.assets.base_path` option.

**Comment by Alsatian67** [comment]
@Cordoval, to reproduce the error, you must only set a path like "/source" by framework.templating.assets_base_urls.

@xabbuh 
Ok, with framework.assets.base_path I can recreate what I want.

The with sf < 2.7.BETA-1, it was possible to define a route relative to the web path, now it is not possible anymore, isn't it a BC break ?

**Comment by stof** [comment]
@Alsatian67 passing a path rather than an url for ``assets_base_urls`` was not a supported use case though

**Comment by Alsatian67** [comment]
It worked fine by me with every Sf versions from 2.2 to 2.6.
I used it in all my apps to seperate prod and dev assets, and point to the good path.

Then there was a bug before Sf 2.7 :

A path by this parameter worked despite it should not work !

**Comment by cordoval** [comment]
@Alsatian67 @javiereguiluz did a PR this morning https://github.com/symfony/symfony-docs/pull/5170/files but I suggested this to be documented on the UPGRADE2.7.md document. I think would this had been there you wouldn't have had this problem. However as @stof  is mentioning you should pass an url, even this is clear in the body of tests.

**Comment by Alsatian67** [comment]
OK, my problem just came from my wrong understanding of the previous configuration in templating.
I was just wondering that it worked with paths before. If I never should use paths on this parameter, it is not a BC brake.

You can close this issue.

Thank you to every one.

**Comment by mrbase** [comment]
I have the same problem but in packages:

> framework.templating.packages.some_package.base_urls: ''

yields the same error as described above.

We set the base url to "" in development but to the correct cdn url in production. 
This solution worked in sf 2.6 but not in 2.7.

**Comment by xabbuh** [comment]
@mrbase Well, the empty string is no valid URL.

**Comment by mrbase** [comment]
@xabbuh yeah, that is obvious - bit it still breaks bc

**Comment by xabbuh** [comment]
@mrbase As explained in https://github.com/symfony/symfony/issues/14332#issuecomment-92311039, that use case was not supported. I see that this is a bit inconvenient for you, but I don't see anything we can do about that.

**Comment by hjr3** [comment]
I just ran into this as well. I used `""` pre-production and specified a CDN host in production. I fixed it by moving the `base_url` settings into `app/config/prod/framework.yml`.

I think some symfony devs have a misguided view of BC. Whether or not it was officially supported, symfony 2.7 stopped accepting `""` as a valid input to a public facing API. That is breaking BC. I have no problem that the decision was made to break BC. However, I do think that the symfony could have done a better job of communicating this out and educating people on why the decision was made and how to fix it.
